webapp/channels/src/components/team_settings/team_access_tab/team_access_tab.tsx

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useState} from 'react';
+import {useIntl} from 'react-intl';
+
+import ModalSection from 'components/widgets/modals/components/modal_section';
+import SaveChangesPanel, {type SaveChangesPanelState} from 'components/widgets/modals/components/save_changes_panel';
+
+import AllowedDomainsSelect from './allowed_domains_select';
+import InviteSectionInput from './invite_section_input';
+import OpenInvite from './open_invite';
+
+import type {PropsFromRedux, OwnProps} from '.';
+
+import './team_access_tab.scss';
+
+const generateAllowedDomainOptions = (allowedDomains?: string) => {
+    if (!allowedDomains || allowedDomains.length === 0) {
+        return [];
+    }
+    const domainList = allowedDomains.includes(',') ? allowedDomains.split(',') : [allowedDomains];
+    return domainList.map((domain) => domain.trim());
+};
+
+type Props = PropsFromRedux & OwnProps;
```

Comment:
It seems that he index files and the props from redux are only to get the actions, which you can directly use here by using the `useDispatch`. What do you think about renaming this to `index.tsx` and remove the other file?